concent_api/concent_api/middleware.py

```diff
@@ -39,3 +50,80 @@ def __call__(self, request):
         response = self.get_response(request)
         response['Concent-Version'] = self._concent_version
         return response
+
+
+def should_use_html(request):
+    """
+    Function inspects request headers and determines is response should be in HTML format.
+    1) By default it HTML should not be used
+    2) If `text/html` is present in 'Accept' header and 'application/json' is not, HTML should be used
+    3) If both are present, HTML will be used if it has greater weight than JSON
+    """
+    def get_probability(selected_method, accepted_methods):
+        for method in accepted_methods:
+            if selected_method in method:
+                if ":q=" in method:
+                    return float(method.split(":q=")[1])
```

Comment:
Could you maybe make a regex for this whole part. I think it's a bit brittle. For example you're not checking where the `selected_method` is in the string. The current code would probably accept something like `WTF;q=text/html` and fail trying to convert `'text/html'` to `float`.

When constructing the regex you can just assume that `selected_method` contains only alphanumeric characters and `/` (but please `assert` that).

Reply:
there is no point in asserting that `selected_method` (now `selected_content_type`) contains only alphanumeric characters - `get_probability` (`get_weight`)

Reply:
Of course there is - if you implement my suggestion and create a regex.

If I see a regex with an embedded variable like this, it looks like a big, gaping hole in your code. You can force me to analyze the code around it and waste time finding out that it's not possible for anything bad to happen here. Or you can just add a comment explaining that there's nothing to worry about because the only possible values can't have any special characters. Or - even better - you can add an `assert` which won't easily go out of date like a normal comment.